wpuf-functions.php

```diff
@@ -3138,18 +3146,42 @@ function wpuf_encryption( $id ) {
  *
  * @param string $id
  *
- * @return string $encoded_id
+ * @return string|bool decrypted string or false if decryption failed
  */
-function wpuf_decryption( $id ) {
-    $secret_key     = AUTH_KEY;
-    $secret_iv      = AUTH_SALT;
+function wpuf_decryption( $id, $nonce = null ) {
+    // get auth keys
+    $auth_keys = WPUF_Encryption_Helper::get_encryption_auth_keys();
+    if ( empty( $auth_keys ) ) {
+        return false;
+    }
 
-    $encrypt_method = 'AES-256-CBC';
-    $key            = hash( 'sha256', $secret_key );
-    $iv             = substr( hash( 'sha256', $secret_iv ), 0, 16 );
-    $decoded_id     = openssl_decrypt( base64_decode( $id ), $encrypt_method, $key, 0, $iv );
+    $secret_key = $auth_keys['auth_key'];
+    $secret_iv  = ! empty( $nonce ) ? base64_decode( $nonce ) : $auth_keys['auth_salt'];
 
-    return $decoded_id;
+    // should we use sodium_crypto_secretbox_open
+    if ( function_exists( 'sodium_crypto_secretbox_open') ) {
+        try {
+            return sodium_crypto_secretbox_open( $id, $secret_iv, $secret_key );
```

Comment:
$id should be base64_decoded eg: `sodium_crypto_secretbox_open( base64_decode( $id ), $secret_iv, $secret_key );`